src/IO/AzureBlobStorage/copyAzureBlobStorageFile.cpp

```diff
@@ -347,11 +347,12 @@ void copyAzureBlobStorageFile(
     const String & dest_blob,
     std::shared_ptr<const AzureBlobStorage::RequestSettings> settings,
     const ReadSettings & read_settings,
+    bool same_credentials,
     ThreadPoolCallbackRunnerUnsafe<void> schedule)
 {
     auto log = getLogger("copyAzureBlobStorageFile");
 
-    if (settings->use_native_copy)
+    if (settings->use_native_copy && same_credentials)
```

Comment:
We can improve logging a bit.

When we investigate incidents it's usually better to see in the logs
```
"Copying Blob: {} from Container: {}, native copy is disabled"
```
or
```
Copying Blob: {} from Container: {}, native copy is disabled because of different credentials
```
instead of
```
"Reading from Container: {}, Blob: {}"
```

Reply:
addressed it